include/input_packet.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+struct InputPacket {
+    enum InputType { KEYBOARD, MOUSE, KEYBOARD_AND_MOUSE } type;
+    union {
+        struct {
+            int keyCode;  // Represents the key pressed
+            bool pressed;  // True if key is pressed, false if released
+        } keyboard;
+        struct {
+            int x, y;  // Mouse coordinates
+            int button;  // Button pressed
+            bool pressed;  // True if button is pressed, false if released
+        } mouse;
+    };
+
+    void serialize(const InputPacket& input, char*& outData);
+    void deserialize(const char* inData, InputPacket& input);
```

Comment:
Can these be static if the caller isn't needed? i.e., if we call using `somePacket.serialize(packetToSerialize, outData);`, do we need `somePacket` to be tied to the operation?